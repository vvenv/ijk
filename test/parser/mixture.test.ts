import { expect, test } from 'vitest';
import { parse } from './__helper';

test('for -> if', () => {
  expect(
    parse(
      `{{ for name in names }}{{ if name }}{{ name }}{{ endif }}{{ endfor }}`,
    ),
  ).toMatchSnapshot();
});

test('if -> for', () => {
  expect(
    parse(
      `{{ if names }}{{ for name in names }}{{ name }}{{ endfor }}{{ endif }}`,
    ),
  ).toMatchSnapshot();
});

test('if -> for -> if', () => {
  expect(
    parse(
      `{{ if name }}{{ for name in names }}{{ if name }}{{ name }}{{ endif }}{{ endfor }}{{ endif }}`,
    ),
  ).toMatchSnapshot();
});

test('for -> if/else ', () => {
  expect(
    parse(
      `{{ for name in names }}{{ if name }}{{ name }}{{ else }}***{{ endif }}{{ endfor }}`,
    ),
  ).toMatchSnapshot();
});

test('for -> if/else - literal', () => {
  expect(
    parse(
      `{{ for name in names }}{{ if name }}{{ name }}{{ else }}{{ "***" }}{{ endif }}{{ endfor }}`,
    ),
  ).toMatchSnapshot();
});
