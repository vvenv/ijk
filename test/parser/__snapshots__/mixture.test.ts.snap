// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`for -> if 1`] = `""use strict";let s="";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for(let i_1_0=0;i_1_0<l_1_0;i_1_0++){const c_i_1_0={...c,name:c.names[k_1_0[i_1_0]],loop:{index:i_1_0,first:i_1_0===0,last:i_1_0===l_1_0,length:l_1_0}};if(c_i_1_0.name){s+=e(c_i_1_0.name);}}return s;"`;

exports[`for -> if/else  1`] = `""use strict";let s="";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for(let i_1_0=0;i_1_0<l_1_0;i_1_0++){const c_i_1_0={...c,name:c.names[k_1_0[i_1_0]],loop:{index:i_1_0,first:i_1_0===0,last:i_1_0===l_1_0,length:l_1_0}};if(c_i_1_0.name){s+=e(c_i_1_0.name);}else{s+="***";}}return s;"`;

exports[`for -> if/else - literal 1`] = `""use strict";let s="";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for(let i_1_0=0;i_1_0<l_1_0;i_1_0++){const c_i_1_0={...c,name:c.names[k_1_0[i_1_0]],loop:{index:i_1_0,first:i_1_0===0,last:i_1_0===l_1_0,length:l_1_0}};if(c_i_1_0.name){s+=e(c_i_1_0.name);}else{s+=e("***");}}return s;"`;

exports[`if -> for -> if 1`] = `""use strict";let s="";if(c.name){const k_2_0=Object.keys(c.names);const l_2_0=k_2_0.length;for(let i_2_0=0;i_2_0<l_2_0;i_2_0++){const c_i_2_0={...c,name:c.names[k_2_0[i_2_0]],loop:{index:i_2_0,first:i_2_0===0,last:i_2_0===l_2_0,length:l_2_0}};if(c_i_2_0.name){s+=e(c_i_2_0.name);}}}return s;"`;

exports[`if -> for 1`] = `""use strict";let s="";if(c.names){const k_2_0=Object.keys(c.names);const l_2_0=k_2_0.length;for(let i_2_0=0;i_2_0<l_2_0;i_2_0++){const c_i_2_0={...c,name:c.names[k_2_0[i_2_0]],loop:{index:i_2_0,first:i_2_0===0,last:i_2_0===l_2_0,length:l_2_0}};s+=e(c_i_2_0.name);}}return s;"`;
